unittests/data/type_traits.hpp

```diff
@@ -430,6 +425,15 @@ namespace before{
     typedef const incomplete_type x2;
     typedef int* const x3;
     typedef int* volatile x4;
+    typedef void const * x5;
+
+    typedef char arr_42[42];
+    typedef char const arr_c_42[42];
+    typedef char volatile arr_v_42[42];
+#ifdef __castxml__
+    typedef char const volatile arr_cv_42[42];
+    typedef char volatile const arr_vc_42[42];
+#endif
```

Comment:
I figured out why this is failing for gccxml, don't have a solution yet.

The XML generated by gccxml for `arr_cv_42` is

```
  <ArrayType id="_164" min="0" max="41u" type="_161" size="336" align="8"/>
  <Typedef id="_165" name="arr_cv_42" type="_164cv" context="_160" location="f1:433" file="f1" line="433"/>
  <FundamentalType id="_161" name="char" size="8" align="8"/>
  <CvQualifiedType id="_164cv" type="_164" const="1" volatile="1"/>
```
If you start with the `Typedef` and follow the `type` attributes, the order is 
```
Typedef -> CvQualifiedType -> ArrayType -> FundamentalType
```

The corresponding XML generated by CastXML is

```
  <Typedef id="_16" name="arr_cv_42" type="_19" context="_11" location="f1:433" file="f1" line="433"/>
  <ArrayType id="_19" min="0" max="41" type="_24cv"/>
  <FundamentalType id="_24" name="char" size="8" align="8"/>
  <CvQualifiedType id="_24cv" type="_24" const="1" volatile="1"/>
```
In this case, the order is 
```
Typedef -> ArrayType -> CvQualifiedType -> FundamentalType
```

`remove_const` in *type_traits.py* expects `const_t` and/or `volatile_t` to be wrapped within `array_t`, and not the other way around, so it fails to remove `const_t` for GCCXML.

At this point, I'm unsure whether the right fix is to handle the two different orders within `remove_const` or to re-order the types for GCCXML. The problem with the first solution is we might run into this issue for other type traits too, but I have no idea right now how complicated the second solution is.